src/event/reader.rs

```diff
@@ -686,12 +764,16 @@ impl ReaderState {
     async fn wait_for_segment_slice_return(&mut self, segment: &ScopedSegment) -> Option<SliceMetadata> {
         if let Some(receiver) = self.slice_release_receiver.remove(segment) {
             match receiver.await {
+                //TODO: Handle error here.
```

Comment:
Returning a none incase of an Error should be fine here. (Testing of this part is in progress).